jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/core/JakartaForJavaAssert.java

```diff
@@ -156,22 +156,32 @@ public static Range r(int startLine, int startCharacter, int endLine, int endCha
     public static Position p(int line, int character) {
         return new Position(line, character);
     }
+  
+    public static void assertJavaDiagnosticsPresent(JakartaDiagnosticsParams params, JDTUtils utils, Diagnostic... expected)
+            throws JavaModelException {
+        List<PublishDiagnosticsParams> actual = JDTServicesManager.getInstance().getJavaDiagnostics(params);
 
+        assertDiagnostics(true, 
+                actual != null && actual.size() > 0 ? actual.get(0).getDiagnostics() : Collections.emptyList(),
+                expected);
+    }
+    
     public static void assertJavaDiagnostics(JakartaDiagnosticsParams params, JDTUtils utils, Diagnostic... expected)
             throws JavaModelException {
         List<PublishDiagnosticsParams> actual = JDTServicesManager.getInstance().getJavaDiagnostics(params);
 
-        assertDiagnostics(
+        assertDiagnostics(false, 
                 actual != null && actual.size() > 0 ? actual.get(0).getDiagnostics() : Collections.emptyList(),
                 expected);
     }
 
-    public static void assertDiagnostics(List<Diagnostic> actual, Diagnostic... expected) {
-        assertDiagnostics(actual, Arrays.asList(expected), false);
+    public static void assertDiagnostics( boolean checkForDiagnostic, List<Diagnostic> actual, Diagnostic... expected) {
+        assertDiagnostics(actual, Arrays.asList(expected), false, checkForDiagnostic);
     }
 
-    public static void assertDiagnostics(List<Diagnostic> actual, List<Diagnostic> expected, boolean filter) {
-        /**
+    public static void assertDiagnostics(List<Diagnostic> actual, List<Diagnostic> expected, boolean filter, boolean isDiagnosticPresent) {
```

Comment:
It is confusing when the argument is called checkForDiagnostic in one method and is called something different here. I suggest you rename this argument to `checkForDiagnostic`.